source/common/ur_util.hpp

```diff
@@ -87,23 +91,120 @@ inline std::string create_library_path(const char *name, const char *path) {
 #endif
 
 ///////////////////////////////////////////////////////////////////////////////
-inline bool getenv_tobool(const char *name) {
-    const char *env = nullptr;
-
+inline std::optional<std::string> ur_getenv(const char *name) {
 #if defined(_WIN32)
-    char buffer[8];
-    auto rc = GetEnvironmentVariable(name, buffer, 8);
-    if (0 != rc && rc <= 8) {
-        env = buffer;
+    constexpr int buffer_size = 1024;
+    char buffer[buffer_size];
+    auto rc = GetEnvironmentVariable(name, buffer, buffer_size);
+    if (0 != rc && rc < buffer_size) {
+        return std::string(buffer);
+    } else if (rc >= buffer_size) {
+        std::cerr << "Environment variable " << name << " value too long!"
+                  << " Maximum length is " << buffer_size - 1 << " characters." << std::endl;
     }
+    return std::nullopt;
 #else
-    env = getenv(name);
+    const char *tmp_env = getenv(name);
+    if (tmp_env != nullptr) {
+        return std::string(tmp_env);
+    } else {
+        return std::nullopt;
+    }
 #endif
+}
+
+inline bool getenv_tobool(const char *name) {
+    auto env = ur_getenv(name);
+    return env.has_value();
+}
+
+static void wrong_format_vec_msg(const char *env_var_name) {
+    std::cerr << "Wrong format of the " << env_var_name << " environment variable!"
+              << " Proper format is: ENV_VAR=\"value_1,value_2,value_3\"" << std::endl;
+}
+
+static void wrong_format_map_msg(const char *env_var_name) {
+    std::cerr << "Wrong format of the " << env_var_name << " environment variable!"
+              << " Proper format is: ENV_VAR=\"param_1:value_1,value_2;param_2:value_1" << std::endl;
+}
+
+inline std::optional<std::vector<std::string>> getenv_to_vec(const char *env_var_name) {
```

Comment:
please add briefs on what these two functions do. It's hard to get from just the names.